tests/Examples/tfhe_rust_hl/fpga/README.md

```diff
@@ -0,0 +1,47 @@
+# End to end Rust codegen tests - High Level API
+
+These tests generate Rust code for the
+[tfhe-rs](https://github.com/zama-ai/tfhe-rs) backend library, including
+compiling the generated Rust source and running the resulting binary on Belfort
+FPGAs. These target the integer plaintexts and the accompanying library.
+
+To avoid introducing these large dependencies into the entire project,
+tests are manual, and require the system they're running on to have
+[Cargo](https://doc.rust-lang.org/cargo/index.html) installed. During the test,
+cargo will fetch and build the required dependencies.
+
+## Prerequisites
+
+Install the heir toolchain from the nightly release:
+
+```bash
+wget https://github.com/google/heir/releases/download/nightly/heir-translate
+chmod +x heir-translate
+```
+
+## FPGA Backend
+
+The FPGA backend is built using AWS EC2 F2 instances. They are configured with the Belfort Amazon Machine Image (AMI). Instructions for building the AMI can be found on [Belfort's Github](https://github.com/belfortlabs/hello-fpga).
+
+Cargo.toml depends on a `tfhe-rs` fork that contains the set-up for the FPGA backend. The following 3 lines of code were added to `main.rs` to connect to the FPGA:
+
+```rust
+let mut fpga_key = BelfortServerKey::from(&server_key);
+fpga_key.connect();
+set_server_key(fpga_key.clone());
+```
+
+## Running the example
+
+To transpile the .mlir file to rust code, run the following command:
+
+```bash
+cd heir/tests/Examples/tfhe_rust_hl/fpga
+~/heir-translate ./arith.mlir --emit-tfhe-rust-hl > ./src/fn_under_test.rs
```

Comment:
We could also provide a bazel command here instead of using the nightly:

```
bazel run //tools:heir-translate -- --emit-tfhe-rust-hl $PWD/arith.mlir > $PWD/src/fn_under_test.rs
```

Reply:
Using bazel requires an extra tool as well as a correct clang version. For me downloading the binary was more straightforward. I'm happy to include the bazel command if you want, but following the KISS principle, I would exclude it.

Reply:
Maybe we can make the pre-reqs generic and link to https://heir.dev/docs/getting_started/#getting-heir to get `heir-translate` either from nightly / built from source.

Then we can just use `heir-translate` generically below to indicate use.

Reply:
I agree the nightly is simpler setup. This command would be needed to test local changes to HEIR. Though I suppose someone who is already making local changes to HEIR would not have trouble constructing the right bazel command. It's probably fine.